src/Microsoft.NetCore.Analyzers/UnitTests/Security/UseAutoValidateAntiforgeryTokenTests.cs

```diff
@@ -0,0 +1,1369 @@
+        {
+            VerifyCSharpWithDependencies(@"
+using System;
+using Microsoft.AspNetCore.Mvc;
+
+[MyValidateAntiForgeryAttribute]
+class TestClass : ControllerBase
+{
+    [HttpPost]
+    public AcceptedAtActionResult SubAcceptedAtAction (string actionName)
+    {
+        return null;
+    }
+}");
+        }
+
+        [Fact]
+        public void Test_NotUsingValidateAntiForgeryAttribute_NoDiagnostic()
+        {
+            VerifyCSharpWithDependencies(@"
+using Microsoft.AspNetCore.Mvc;
+
+class TestClass : ControllerBase
+{
+    [HttpPost]
+    public override AcceptedAtActionResult AcceptedAtAction (string actionName)
+    {
+        return null;
+    }
+}");
+        }
+
+        [Fact]
+        public void Test_ActionMethodWithHttpGetAttribute_NoDiagnostic()
+        {
+            VerifyCSharpWithDependencies(@"
+using Microsoft.AspNetCore.Mvc;
+
+[MyValidateAntiForgeryAttribute]
+class MakeSureValidateAntiForgeryAttributeIsUsedSomeWhereClass : ControllerBase
+{
+}
+
+class TestClass : ControllerBase
+{
+    [HttpGet]
+    public override AcceptedAtActionResult AcceptedAtAction (string actionName)
+    {
+        return null;
+    }
+}");
+        }
+
+        [Fact]
+        public void Test_ValidateAntiForgeryTokenAttributeOnActionMethod_ActionMethodMissingHttpVerbAttribute_NoDiagnostic()
+        {
+            VerifyCSharpWithDependencies(@"
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Antiforgery.Internal;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+class TestClass : ControllerBase
+{
+    [MyValidateAntiForgeryAttribute]
+    public override AcceptedAtActionResult AcceptedAtAction (string actionName)
+    {
+        return null;
+    }
+}");
+        }
+
+        [Fact]
+        public void Test_ValidateAntiForgeryTokenAttributeOnController_ActionMethodMissingHttpVerbAttribute_NoDiagnostic()
+        {
+            VerifyCSharpWithDependencies(@"
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Antiforgery.Internal;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+[MyValidateAntiForgeryAttribute]
+class TestClass : ControllerBase
+{
+    public override AcceptedAtActionResult AcceptedAtAction (string actionName)
+    {
+        return null;
+    }
+}");
+        }
+
+        [Fact]
+        public void Test_NoValidateAntiForgeryTokenAttribute_ActionMethodMissingHttpVerbAttribute_NoDiagnostic()
+        {
+            VerifyCSharpWithDependencies(@"
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Antiforgery.Internal;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+class TestClass : ControllerBase
+{
+    public override AcceptedAtActionResult AcceptedAtAction (string actionName)
+    {
+        return null;
+    }
+}");
+        }
```

Comment:
I think this case should produce a diagnostic.  No global antiforgery and no antiforgery token, and the action method will accept HTTP POSTs since there's no [HttpMethod] attribute.